Sources/MongoSwift/ConnectionString.swift

```diff
@@ -4,6 +4,8 @@ import CLibMongoC
 internal class ConnectionString {
     /// Pointer to the underlying `mongoc_uri_t`.
     private let _uri: OpaquePointer
+    /// Tracks whether we have already destroyed the above pointer.
+    private var destroyedPtr = false
```

Comment:
I considered going with a `State` enum approach. however, this would have required wide-sweeping changes throughout the type since we use `self._uri` _a lot_, including in various computed properties where we can't throw. 

Given that the only time we manually destroy the pointer is in `init` (and it's only in a case where the init throws, and therefore `self` doesn't exist anymore), it's not possible for us to accidentally use a destroyed `self._uri` anywhere else in the type, so it seems fairly low risk to go with this approach.

open to any other opinions or ideas you have though, I figured I would just start out with the simplest solution.